pages/kids/image-quiz/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { useTranslation } from 'next-i18next';
+import React, { useEffect, useState } from 'react';
+import Image from 'next/image';
+import { Phrase_deprecated, TranslationJSON } from 'utils/Phrase_deprecated';
+import { getCountryVariant } from 'utils/locales';
+import SEO from 'components/basecomponents/SEO';
+import { GetStaticProps, InferGetStaticPropsType } from 'next';
+import { DictionaryDataObject, fetchDictionary, getKidsCategory, Phrase } from '../../../utils/getDataUtils';
+import { getServerSideTranslations } from '../../../utils/localization';
+import { useLanguage } from 'utils/useLanguageHook';
+import { KidsTranslation } from 'components/basecomponents/KidsTranslation';
+import { AudioPlayer } from 'utils/AudioPlayer';
+import phrases_CS from 'components/basecomponents/MemoryGame/memory-game-cs.json';
+import phrases_PL from 'components/basecomponents/MemoryGame/memory-game-pl.json';
+import phrases_SK from 'components/basecomponents/MemoryGame/memory-game-sk.json';
+import styles from './imageStyle.module.css';
+
+export type KidsTranslation = TranslationJSON & { image: string };
+
+interface ImageContainerProps {
+  phrase: Phrase;
+  onClick: (e: React.MouseEvent, phrase: Phrase, correct: boolean) => void;
+  correct: boolean;
+}
+
+const CHOICES_COUNT = 3;
+
+const getRandomIndex = (len = CHOICES_COUNT) => Math.floor(Math.random() * len);
+
+const GAME_NARRATION_PHRASES = {
+  cs: phrases_CS,
+  sk: phrases_SK,
+  pl: phrases_PL,
+};
+
+const matchSound = '/kids/image-quiz/positive.mp3';
+const dontMatchSound = '/kids/image-quiz/negative.mp3';
+const text = GAME_NARRATION_PHRASES[getCountryVariant()];
+
+const ImageQuizSection = ({ dictionary }: InferGetStaticPropsType<typeof getStaticProps>) => {
+  const [phrases, setPhrases] = useState<Phrase[]>();
+  const [correctIndex, setCorrectIndex] = useState(getRandomIndex);
+  const { t } = useTranslation();
+  const { currentLanguage, otherLanguage } = useLanguage();
+
+  useEffect(() => {
+    const _phrases: Phrase[] = shuffle(getKidsCategory(dictionary)?.translations);
+    setPhrases(_phrases);
+  }, [dictionary]);
+
+  useEffect(() => {
```

Comment:
`useEffect` is not the best tool for this here. It works now, but imagine you had another piece of state that was not related to loading new pictures. It would fire this effect and make the game out of sync. The quick fix is to specify a dependency array `[correctIndex, otherLanguage, phrases]`. 

The actual fix is to move this code into the `handleClick` event handler, consider doing that if you feel like it :)

Reply:
handlerClick has setPhrases, this useEffect supposed to shuffle images on mount, with empty array, i can add eslint ignore.

Reply:
Ok, thank you, I will move it into handlerClick.